src/containers/task-group.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Task from '../components/task';
+
+class TaskGroup extends Component {
+
+  processTaskList(list, givenMonth, givenType) {
+    let resultsArray = [];
+    const loop = list.map( (item) => {
+      if(item.month === givenMonth && item.type === givenType) {
+        resultsArray.push(item);
+      };
+    });
+    if(resultsArray.length) {
+      return true;
+    }
+  }
+
+  taskTypeIsUsed(list, givenMonth, givenType) {
+    let taskIsUsed = false;
+
+    // verification is an array containing lots of
+    // 'undefined' + occasionally true boolean
+    const verification = list.map( (item) => {
+      if(this.processTaskList(item.tasks, givenMonth, givenType) !== undefined) {
+        taskIsUsed = true;
+      }
+    });
+
+    return taskIsUsed;
+  };
+
+  taskHasItems(list, givenMonth, givenType) {
+    return this.processTaskList(list, givenMonth, givenType);
+  };
+
+  renderTask(taskType) {
+    const taskClass = taskType.toLowerCase().replace(" ", "-");
+
+    if(this.taskTypeIsUsed(this.props.data, this.props.thisMonth, taskType)) {
+      return (
+        <li className={`task task--${taskClass}`}>
+          <h3 className={`task__title task__title--${taskClass}`}>{taskType}</h3>
+          <ul className="task__item-list">
+            {this.renderTaskItems(taskType)}
+          </ul>
+        </li>
+      );
+    }
+  };
+
+  renderTaskItems(taskType) {
+    return this.props.data.map( (item) => {
```

Comment:
Method can be simplified to:

    return this.props.data
        .filter(item => this.taskHasItems(item.tasks, this.props.thisMonth, taskType))
        .map(item => <Task variety={item.variety} name={item.name}) />)

(JSX, not sure if you need parens on the tag in the map.)

Technically it gives a small performance loss (loops over the array twice, unless the compiler is smart enough to optimise it out), but that shouldn't really be an issue, and it's much more succinct.